controllers/toolchainstatus/toolchainstatus_controller.go

```diff
@@ -479,6 +480,16 @@ func getAPIEndpoint(clusterName string, memberClusters []*cluster.CachedToolchai
 	return ""
 }
 
+func fetchDomainFromUrl(logger logr.Logger, proxyUrl string) string {
+	url, err := url.Parse(proxyUrl)
+	if err != nil {
+		logger.Error(err, fmt.Sprintf("Error while parsing the proxyURL"))
```

Comment:
should we return the error here , as it might not be able to proceed with the rest of the function?